### ISSUE TITLE
Deployment option

### ISSUE THREAD

**Issue by ArmanFalah** [otykier/TabularEditor#196]
Hi Daniel,

I hope you are doing well.

It is perfect that we can set a backup address in Tabular Editor. I think it will be great if you add a password to this zip files. You can add another box for the password in deployment option and then the backup files need the password to open.

Regards Arman

![tabular editor](https://user-images.githubusercontent.com/24508605/46578948-d0bd7000-ca15-11e8-8942-1d7f27ad283a.png)

**Comment by otykier** [comment]
Hi Arman
Glad to hear you're using that feature :-)
Sure, I can add two checkboxes to the preferences dialog:

1. Zip backup files
2. Protect with password

By default, both will be unchecked, and option 2 is only enabled when option 1 is checked. When both are checked, you will be prompted during deployment to set the password of the zip file. Would that work for you?

**Comment by ArmanFalah** [comment]
of course, it will be great. 👍 
but I didn't get it why do you want to add zip backup files checkbox. the backup files are zip files now.

**Comment by otykier** [comment]
Oh, haha - I completely forgot about that :-)
Will just have the checkbox for whether or not they should be password protected then!

**Comment by ArmanFalah** [comment]
:))
I think just need a text box to input the Password for zip files. Don't you think so?

**Comment by otykier** [comment]
Yes, but it's probably best to prompt for that during deployment - otherwise Tabular Editor would have to store the password somewhere, and that's something I'd like to avoid.

**Comment by ArmanFalah** [comment]
you mean every time we want to deploy Tabular editor prompt for that?

**Comment by otykier** [comment]
Yes, that was the idea. Do you think that's bad?

**Comment by otykier** [comment]
Wait a second - are you using the Deploy option for that? I think you should adopt a different workflow in that case. The "Deploy" option is intended to be used when you want to deploy your model to a different environment - for example, a production server.

For development, the ideal workflow is the following:

1. Open your Model.bim file (Ctrl+O) in Tabular Editor and use the Deploy option to put it on your **development** server.
2. Open the database from the development server (Ctrl+Shift+O) in Tabular Editor.
3. Make your changes. Hit Save (Ctrl+S) whenever you want to apply the changes to the database.
4. When you are done working, use Deploy again to put the changes on your **test** or **production** server.
5. Use "Save As..." to save the changes back to the original Model.bim file, so that you can commit the changes to your source control environment (if you have that).

If you don't have separate servers for development/test/production, I suggest you simply create multiple copies of the database on the same server, to have at least one database for development and another for production.

This workflow has the benefit that you don't need to go through the Deployment Wizard UI every time you want to test a small change, and also, Tabular Editor will report any DAX syntax errors reported from the server, and show them directly in the UI:

<img width="251" alt="screen shot 2018-10-08 at 11 37 38" src="https://user-images.githubusercontent.com/8976200/46601800-9542a480-caee-11e8-9649-352bbff6801e.png">

Let me know if you have any questions.

**Comment by ArmanFalah** [comment]
yes, that's completely right that we should have separate servers.
I think there is a misunderstanding here, Daniel.
when I am saying "Deployment option" what I mean is the deployment option page in Tabular editor preference. I do not deploy the model. imagine that I work on a test SSAS Server. I open a Tabular model from an existing database. I change some measure and then I saved the model and Tabular Changed the SSAS model and it saves a zip backup file in the address. This is my case. 
![image](https://user-images.githubusercontent.com/24508605/46607584-5a506900-cb0e-11e8-8350-fd11c8231dca.png)

**Comment by otykier** [comment]
Ah, thanks for that clarification. It's been so long since I've used this feature, I completely forgot that the .zip files are created also when you save the model :-)

In that case, what I propose is the following: If you check the "Protect with password" option, as I suggested above, then Tabular Editor will only prompt you for a password the *first time* you hit save, storing the password in memory and reusing it the next time you save - at least until you exit Tabular Editor. If the password should be persisted between sessions of Tabular Editor, I have to store it securely somewhere in a file or the registry, and that's something I'd like to avoid, to be honest.
